thunder/__init__.py

```diff
@@ -814,7 +814,50 @@ def maybe_call_epilogue(cache_entry, result, pro_to_epi):
 
         return result
 
+    def unwrap_inner_exception(c: Callable) -> Callable:
+        def _thunder_unwrap_inner_exception(*args, **kwargs):
+            # Run the function, and caputre the exception if there is one.
+            try:
+                return c(*args, **kwargs)
+            except InnerException as e:
+                exc = e.value
+
+            def internal_to_thunder(co):
+                if co is thunder_general_jit.__code__ or co is _thunder_unwrap_inner_exception.__code__:
+                    return True
+                return co.co_filename.endswith("thunder" + os.sep + "core" + os.sep + "interpreter.py") and (
+                    co.co_name in ("fn_", "fn_2")
+                )
+
+            # Iterate over the traceback and collect frames that don't correspond to thunder internal functions.
+            tb = exc.__traceback__
+            tb_frames = []
+            while tb != None:
+                co = tb.tb_frame.f_code
+                if not internal_to_thunder(co):
+                    tb_frames.append(tb)
+                tb = tb.tb_next
+
+            # Relink the non-internal traceback frames
+            if tb_frames:
+                top_tb = tb = tb_frames[0]
+                for _tb in tb_frames[1:]:
+                    tb.tb_next = _tb
+                    tb = _tb
+                exc.__traceback__ = top_tb
+
+            # Re-raise the exception without retaining it in this stack frame to avoid leaking tensors.
+            try:
+                raise exc
+            except Exception as e:
```

Comment:
e seems unused.

Reply:
I style-matched multiple places in the interpreter where we do this. But you may be correct, I'll try it.